### ISSUE TITLE
Some tests fail when run in qpm

### ISSUE THREAD

**Issue by dyfer** [supercollider/supercollider#5360]
## Environment

* SuperCollider version: current develop
* Operating system: Linux, macOS
* Other details (Qt version, audio driver, etc.): test suite run with qpm

## Steps to reproduce

Instructions from https://github.com/supercollider/supercollider/pull/5332#issuecomment-776672228

- install qpm with `pip2`
  - `pip2 install git+https://github.com/supercollider/qpm.git@topic/xvfb-off`
  - note that we are temporarily using the `topic/xvfb-off` branch of qpm
- install API quark 
  - with `python2 /usr/bin/qpm quark checkout API -l <some path>`
    - default checkout path only makes sense on macOS (`~/Library/Application Support`)
  - OR by checking it out `git clone --depth=1 https://github.com/supercollider-quarks/API`
- On Linux, add path with scscyth to the PATH, e.g.
`export PATH=/path/to/supercollider/build/Install/bin:$PATH`
- create a copy of the json file (it will be updated with test results)
`cp /path/to/supercollider/testsuite/scripts/gha_test_run_proto.json /path/to/supercollider/testsuite/scripts/run/gha_test_run.json`
- run with `python2 /usr/bin/qpm test.run -l <json copy> --path <sclang> --include <API path> <testsuite/classlibrary path>`

## Expected vs. actual behavior

Some tests fail only when run in qpm. See #5332 for more information.

When fixing the tests, please remove appropriate entries in [gha_test_run_proto.json](https://github.com/supercollider/supercollider/blob/develop/testsuite/scripts/gha_test_run_proto.json)

Let's use the lists below to keep track of fixed tests.

- [ ] TestBus.test_get: UnitTest sometimes fails when run in qpm
- [ ] TestCondition.*: UnitTest fails when run in qpm
- [ ] TestDocumentEnvironment.*: UnitTest creates issues when run in qpm
- [ ] TestDoneActions.test_freeSelfAndResumeNext: UnitTest fails when run in qpm
- [ ] TestEnvGen_server.test_shapeHold_setEndValue: UnitTest fails when run in qpm on Linux
- [ ] TestEnvGen_server.test_trig_gate_nonsustain_env: UnitTest fails on supernova in qpm on Linux
- [ ] TestEnvironmentDocument.*: UnitTest fails when run in qpm
- [ ] TestLinkClock.*: UnitTest has unknown issues when run in qpm
- [ ] TestNodeProxyRoles.test_proxyroles_xfade_smoothly: UnitTest fails in qpm
- [ ] TestOSCFunc.test_argSizeMatching_Succeeds: UnitTest fails when run in qpm on Linux
- [ ] TestPV_ChainUGen.*: Server might hang after these tests when run in qpm
- [ ] TestWebView.*: UnitTest sometimes fails when run in qpm on Linux

**Comment by dyfer** [comment]
Note to self and other coming back to this: I was able to reproduce some of these issues by running even individual tests, but on a system that's under a full (artificial) load. To do that on *nix, one can do `yes > /dev/null` in terminal, duplicated as many times as there are cores in the system (both "efficiency" and "performance" cores, where applicable), then run the test in SC. One should be able to reproduce some of the issues semi-reliably that way...

**Comment by JordanHendersonMusic** [comment]
I think we can check off the Document tests, these are impossible to run in a CLI environment because we don't have an open document.

**Comment by JordanHendersonMusic** [comment]
I've ticked off `TestThreadReschedule:test_Threads_rescheduleToOtherClock` as I can't recreate this failure, I think the try fix has solved the issue.

**Comment by dyfer** [comment]
Yes, feel free to tick off tests that were fixed in the meantime, thanks!

**Comment by dyfer** [comment]
> I think we can check off the Document tests, these are impossible to run in a CLI environment because we don't have an open document.

@JordanHendersonMusic given the improvements you're working on, I think it may be at some point possible to run just the Document tests in the IDE, but that is low priority arguably.
